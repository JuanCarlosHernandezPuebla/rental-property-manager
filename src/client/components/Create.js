import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Create extends Component {

    render(){
        return (
            <div className="container">
                <form className="form-signup" method="POST">
                    <img className="mb-4" src="/images/rpm_logo.png" alt="" width="72" height="72" />
                    <h1 className="h3 mb-3 font-weight-normal">Create an account</h1>
                    <label htmlFor="inputUsername">Username</label>
                    <input type="text" id="inputUsername" name="username" className="form-control" placeholder="Username" required autoFocus />
                    <label htmlFor="inputPassword">Password</label>
                    <input type="password" id="inputPassword" name="password" className="form-control" placeholder="Password" required />
                    <button className="btn btn-lg btn-primary btn-block btn-login" type="submit">Create an account</button>
                    <div>
                        <Link className="nav-link" to="/login">Log Into Existing Account</Link>
                    </div>
                </form>
            </div>
        );
    }
}

export default Create;