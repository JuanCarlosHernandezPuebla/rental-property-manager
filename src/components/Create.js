import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { Modal } from 'react-bootstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import SignupForm from './Forms/SignupForm';

export default function Create() {
  const [modal, setModal] = useState({
    show: false,
    icon: '',
    title: '',
    body: ''
  });

  const handleClose = () => setModal({ show: false });

  return (
    <div className="container">
      <div className="signup-form-content">
        <SignupForm setModal={setModal} />
        <div className="text-center">
          <Link className="nav-link" to="/login">Log Into Existing Account</Link>
        </div>
      </div>
      <Modal
        show={modal.show}
        centered
        onHide={handleClose}
      >
        <Modal.Header closeButton>
          <Modal.Title>
            {modal.icon && <FontAwesomeIcon icon={modal.icon} className="mr-2" />} {modal.title}
          </Modal.Title>
        </Modal.Header>
        <Modal.Body>
          {modal.body}
        </Modal.Body>
      </Modal>
    </div>
  );
}
