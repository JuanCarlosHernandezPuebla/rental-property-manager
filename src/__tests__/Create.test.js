import React from 'react';
import { shallow } from 'enzyme';
import Create from '../components/Create';
import renderToJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router'

test('Create render test', () => {
  const component = shallow(
    <MemoryRouter initialEntries={[{ pathname: '/create', key: 'testKey' }]} >
      <Create />
    </MemoryRouter>
  );
  expect(renderToJson(component)).toMatchSnapshot();
});