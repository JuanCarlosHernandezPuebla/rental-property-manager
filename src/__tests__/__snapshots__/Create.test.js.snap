// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create render test 1`] = `
<div
  className="container"
>
  <form
    className="form-signup"
    method="POST"
    onSubmit={[Function]}
  >
    <img
      alt=""
      className="mb-4"
      height="72"
      src="/images/rpm_logo.png"
      width="72"
    />
    <h1
      className="h3 mb-3 font-weight-normal"
    >
      Create an account
    </h1>
    <label
      htmlFor="input_username"
    >
      Username
    </label>
    <input
      autoFocus={true}
      className="form-control"
      id="input_username"
      name="username"
      placeholder="Username"
      required={true}
      type="text"
    />
    <label
      htmlFor="input_password"
    >
      Password
    </label>
    <input
      className="form-control"
      id="input_password"
      name="password"
      placeholder="Password"
      required={true}
      type="password"
    />
    <button
      className="btn btn-lg btn-primary btn-block btn-login"
      type="submit"
    >
      Create an account
    </button>
    <div>
      <Link
        className="nav-link"
        replace={false}
        to="/login"
      >
        Log Into Existing Account
      </Link>
    </div>
  </form>
</div>
`;
